Source/WebInspectorUI/UserInterface/Models/Resource.js

```diff
@@ -435,6 +435,13 @@ WI.Resource = class Resource extends WI.SourceCode
         return this._mimeType;
     }
 
+    get hasMetadata()
+    {
+        // Some metadata is only collected when Web Inspector is open (e.g. resource timing data, HTTP method, request headers, etc.).
+        // Use `requestIdentifier` as a general signal since it is always included when metadata is collected.
+        return !!this.requestIdentifier;
```

Comment:
please use `this._requestIdentifier` since this is inside the same class that declares it (i.e. it's not a layering violation)